ports/atmel-samd/common-hal/analogio/AnalogOut.c

```diff
@@ -31,32 +31,72 @@
 #include "py/runtime.h"
 
 #include "shared-bindings/analogio/AnalogOut.h"
+#include "shared-bindings/microcontroller/Pin.h"
 
-#include "asf/sam0/drivers/dac/dac.h"
-#include "samd21_pins.h"
+#include "atmel_start_pins.h"
+#include "hal/include/hal_dac_sync.h"
+#include "hpl/gclk/hpl_gclk_base.h"
+
+#ifdef SAMD21
+#include "hpl/pm/hpl_pm_base.h"
+#endif
+
+#ifdef SAMD51
+#include "samd51_pins.h"
+#endif
 
 void common_hal_analogio_analogout_construct(analogio_analogout_obj_t* self,
         const mcu_pin_obj_t *pin) {
-    if (pin->pin != PIN_PA02) {
+    if (pin->pin != PIN_PA02
+    #ifdef SAMD51
+        && pin->pin != PIN_PA05
+    #endif
+    ) {
         mp_raise_ValueError("AnalogOut not supported on given pin");
         return;
     }
-    struct dac_config config_dac;
-    dac_get_config_defaults(&config_dac);
-    config_dac.reference = DAC_REFERENCE_AVCC;
-    enum status_code status = dac_init(&self->dac_instance, DAC, &config_dac);
-    if (status != STATUS_OK) {
+
+    self->channel = 0;
+    #ifdef SAMD51
+    if (pin->pin == PIN_PA05) {
+        self->channel = 1;
+    }
+    #endif
+
+    #ifdef SAMD51
+    hri_mclk_set_APBDMASK_DAC_bit(MCLK);
+    hri_gclk_write_PCHCTRL_reg(GCLK, DAC_GCLK_ID, GCLK_PCHCTRL_GEN_GCLK5_Val | (1 << GCLK_PCHCTRL_CHEN_Pos));
```

Comment:
You can use `hri_gclk_set_PCHCTRL_CHEN_bit(const void *const hw, uint8_t index)` instead of the general register write.

Reply:
I'd have to do a second call to set GEN then as well. I copied this out of the ASF4 example code. Still want me to change it?